File: easy/Read-More/Read More.cs
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            // do something with line
            ShowLine(line);
        }
    }
    
    static void ShowLine(string line)
        {
            if (line.Length>55){
                line = line.Substring(0,40);
                if(line.IndexOf(" ")>=0){
                    int i = 39;
                    while(line[i]!=' ')i--;
                    line = line.Substring(0,i);
                }
                line = line + "... <Read More>";
            }
            System.Console.WriteLine(line);
        
        }
}

File: easy/Read-More/Read More.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            ShowLine(line);
        }
    }
    
    static void ShowLine(String line){
        if (line.length()>55){
            line = line.substring(0,40);
            if(line.indexOf(' ')>=0){
                int i = 39;
                while(line.charAt(i)!=' ')i--;
                line = line.substring(0,i);
            }
            line = line + "... <Read More>";
        }
        System.out.println(line);
    }
}
